### PR TITLE
The last server update

### PR DESCRIPTION
**Pull request by alxdarksage**
There may be a few more references to the old terminology here or there, but I've gone through almost every line of code at this point and this is probably good enough for now. These changes introduce the rest of the /v1/apps endpoints and HTTP parameters and similar changes to the API, while keeping all the old stuff for migration.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/services/ExportViaSqsService.java [hunk 1]
```diff
@@ -61,6 +62,7 @@ public void startOnDemandExport(String appId) throws JsonProcessingException {
         ObjectNode requestNode = JSON_OBJECT_MAPPER.createObjectNode();
         requestNode.put(REQUEST_KEY_END_DATE_TIME, endDateTimeStr);
         requestNode.set(REQUEST_KEY_APP_WHITELIST, appWhitelistNode);
+        requestNode.set(REQUEST_KEY_STUDY_WHITELIST, appWhitelistNode);
```
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/services/ExportViaSqsService.java:65]: Do we plan on changing the Exporter to take in the new attribute as well?

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/ParticipantReportController.java [hunk 2]
```diff
@@ -166,7 +166,7 @@ public ReportIndex getParticipantReportIndex(@PathVariable String identifier) {
     }
 
     /** Worker API to get reports for the given user in the given app by date-time. */
-    @GetMapping(path = {"/v1/apps/{appId}/participants/{userId}/reports/{reportId}",
+    @GetMapping(path = {"/v2/apps/{appId}/participants/{userId}/reports/{reportId}",
```
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/spring/controllers/ParticipantReportController.java:169]: Is there any reason we're bumping this to v2 when all the others are in v1?

**Review by DwayneJengSage — APPROVED** [review]
Looks good. Just a few questions.

File: src/main/java/org/sagebionetworks/bridge/services/ExportViaSqsService.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/ExportViaSqsService.java:65]: Yes. The next step is to change all the downstream consumers to use the new API.

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/ParticipantReportController.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/spring/controllers/ParticipantReportController.java:169]: We originally had a /v3/studies/* and /v4/studies/* version of these APIs that took a date and dateTime range, respectively. So I created these as /v1/apps/* and /v2/apps/*.

**Pull request merged**